Sources/Sentry/SentryClient.m

```diff
@@ -634,6 +643,29 @@ - (void)recordLost:(BOOL)eventIsNotATransaction reason:(SentryDiscardReason)reas
     }
 }
 
+- (void)attachScreenshots:(SentryScope *)scope
+{
+#if SENTRY_HAS_UIKIT
+    if (!scope || !self.options.attachScreenshot)
+        return;
+
+    NSArray *shots = [SentryDependencyContainer.sharedInstance.screenshot appScreenshots];
+
+    for (int i = 0; i < shots.count; i++) {
+        NSString *name;
+        if (i == 0)
+            name = @"screenshot.png";
+        else
+            name = [NSString stringWithFormat:@"screenshot-%i.png", i + 1];
+
+        SentryAttachment *att = [[SentryAttachment alloc] initWithData:shots[i]
+                                                              filename:name
+                                                           contentType:@"image/png"];
+        [scope addAttachment:att];
```

Comment:
Does that mean that every following event will duplicate screenshots?
E.g., error 1 has screenshots taken and added to the scope, error 1 is sent with screenshots.
error 2 has screenshots taken and added to the scope, error 2 is sent with screenshots of error 1 and 2?

Reply:
Good point. I thought attachments were cleaned up after being captured. I will check it out.

Reply:
No, they are not cleaned up.

Reply:
We use `hints` on Java, but we don't have hints on Cocoa.